emoji_data_python/replacement.py

```diff
@@ -11,23 +12,29 @@ def replace_colons(text: str, strip: bool=False) -> str:
     >>> emoji_data_python.replace_colons('Hello world ! :wave::skin-tone-3: :earth_africa: :exclamation:')
     'Hello world ! 👋🏼 🌍 ❗'
     """
-    from emoji_data_python import emoji_short_names
+    from emoji_data_python import emoji_short_names, EmojiChar
 
     def emoji_repl(matchobj) -> str:
-        match = matchobj.group(0)
-        codes = match.split(':')
-        res = ''
-        for code in codes:
-            if len(code) > 0:
-                try:
-                    res += emoji_short_names.get(code.replace('-', '_')).char
-                except AttributeError:
-                    if not strip:
-                        res += f':{code}:'
-
-        return res
-
-    return re.sub(r'\:[a-zA-Z0-9-_+]+\:(\:skin-tone-[2-6]\:)?', emoji_repl, text)
+        emoji_match = matchobj.group(1)
+        base_emoji = emoji_short_names.get(emoji_match.strip(':').replace('-', '_'))
+
+        if matchobj.lastindex == 2:
+            skin_tone_match = matchobj.group(2)
+            skin_tone = cast(EmojiChar, emoji_short_names.get(skin_tone_match.strip(':')))
+
+            if base_emoji is None:
+                return f'{emoji_match if strip is False else ""}{skin_tone.char}'
+
+            emoji_with_skin_tone = base_emoji.skin_variations.get(skin_tone.unified)
+            if emoji_with_skin_tone is None:
+                return f'{base_emoji.char}{skin_tone.char}'
+            return emoji_with_skin_tone.char
+        else:
+            if base_emoji is None:
+                return f'{emoji_match if strip is False else ""}'
+            return base_emoji.char
```

Comment:
Else is redundant here as the `if` branch returns in all cases

```suggestion
        
        if base_emoji is None:
            return f'{emoji_match if strip is False else ""}'
        return base_emoji.char
```